src/message.rs

```diff
@@ -0,0 +1,21 @@
+use crate::utils::NodeId;
+use std::net::SocketAddr;
+use serde::{Serialize, Deserialize};
+
+#[derive(Serialize, Deserialize, Debug)]
```

Comment:
Do we want to specify any tagging styles here? https://serde.rs/enum-representations.html